Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/UI/DocumentPropertyEditor/ComponentAdapter.h>
+#include <QtCore/QTimer>
+
+namespace AZ::DocumentPropertyEditor
+{
+    ComponentAdapter::ComponentAdapter()
+    {
+    }
+
+    ComponentAdapter::ComponentAdapter(AZ::Component* componentInstace)
+    {
+        SetComponent(componentInstace);
+    }
+
+    ComponentAdapter::~ComponentAdapter() = default;
+
+    void ComponentAdapter::OnEntityComponentPropertyChanged(AZ::ComponentId componentId)
+    {
+        if (m_componentInstance->GetId() == componentId)
+        {
+            m_refreshQueue++;
+            QTimer::singleShot(
+                0,
+                [this]()
+                {
+                    DoRefresh();
+                });
+        }
+    }
+
+    void ComponentAdapter::InvalidatePropertyDisplay([[maybe_unused]] AzToolsFramework::PropertyModificationRefreshLevel level)
+    {
+        m_refreshQueue++;
+        QTimer::singleShot(
+            0,
+            [this]()
+            {
+                DoRefresh();
+            });
+    }
+
+    void ComponentAdapter::SetComponent(AZ::Component* componentInstance)
+    {
+        m_componentInstance = componentInstance;
+        AzToolsFramework::PropertyEditorEntityChangeNotificationBus::MultiHandler::BusConnect(m_componentInstance->GetEntityId());
+        AzToolsFramework::ToolsApplicationEvents::Bus::Handler::BusConnect();
+        AZ::Uuid instanceTypeId = azrtti_typeid(m_componentInstance);
+        SetValue(m_componentInstance, instanceTypeId);
+    }
+
+    void ComponentAdapter::DoRefresh()
+    {
+        if (m_refreshQueue == 0)
+        {
+            return;
+        }
+        m_refreshQueue = 0;
```

Comment:
The semantics here make me think m_refreshQueue could be a bool. Will there eventually be cases for various levels of queued refreshes?

Reply:
Yeah you're right, this could have been a bool. There aren't any cases right now where we would want anything other than a SoftReset, but I've taken the feedback from Chris and stored the refresh levels in case we want to use them in the future